cmake/compiler-warnings.cmake

```diff
@@ -6,8 +6,14 @@ CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wall)
 
 CHECK_COMPILER_FLAG_AND_ENABLE_IT(-fno-strict-aliasing)
 
-CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wformat)
-CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wformat-security)
+if(WIN32)
+  # MSYS2 gcc compiler gives false positive warnings for (format (printf, 1, 2) - need to turn off for the time being
+  CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wno-format)
+
+  CHECK_COMPILER_FLAG_AND_ENABLE_IT(-mms-bitfields)
+else()
+  CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wformat-security)
```

Comment:
+CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wformat)

Yes, i know that it is part of `-Wall`.